pkg/otlp/config.go

```diff
@@ -79,8 +79,8 @@ func FromAgentConfig(cfg config.Config) (PipelineConfig, error) {
 
 	metricsEnabled := cfg.GetBool(config.OTLPMetricsEnabled)
 	tracesEnabled := cfg.GetBool(config.OTLPTracesEnabled)
-	logsEnabled := cfg.GetBool(config.OTLPLogsEnabled)
-	if !metricsEnabled && !tracesEnabled {
+	logsEnabled := cfg.GetBool(config.OTLPLogsEnabled) || (!cfg.IsSet(config.OTLPLogsEnabled) && HasLogsSection(cfg))
```

Comment:
This looks a bit counter-intuitive, I would rephrase as: if `otlp_config: logs` is present && if `otlp_config: logs: enabled` is not false.